### DISCUSSION TITLE
Deepgram live struggles with proper nouns and other inaccuracies and underperforms wrt to Google transcription services

### DISCUSSION THREAD

**Discussion by diegoasua** [deepgram/community#544] [General help]
Hi,

First of all amazing job at Deepgram. I love the community and the OSS mentality. Docs, playground and examples are A+. As a developer I thrive with companies like Deepgram that make software development a joy ride.

While experimenting with Nova-2 and enhanced models in live `pcm 16 bit`, 16 kHz streams I noticed models struggle with proper nouns. When comparing to Google's live STT it performs worse in most cases I have tested. For example, I said `Amazon` and it understood `Emerson`. I said `James Gorman` and it understood `James Corman`.

Other problems I noted are sometimes it does get proper nouns correctly but simple words around them are simply completely wrong. I asked `how much is Craigslist's revenue` and it understood `for watches craigslist revenue`. I tried again and got `Large cheese cookies, spring meal.`. As you can imagine, I am shocked. This is not just slightly different but completely wrong. When I use Google's live speech to text it gets these right. Just in case there was something wrong with my audio I tried the playground in the landing page  of https://deepgram.com and got `How much is Craigslist Australia?`

In my hands Google performs way beyond Deepgram for live transcription, which is the opposite I expected from the data you show. Since you announce that Deepgram is 53% more accurate than Google, I assume there are some settings I must enable to get this level of accuracy. Is this expected? Is there anything I can do to improve accuracy?

Some more details: I tried with iPhone 14 Pro Max and Mac M2 microphones as well as AirPods as input, no difference. I am in a silent room so there is no background noise.

Thank you

Edit: These are the settings I am using:
```javascript
    const liveTranscription = deepgram.listen.live({
        language: 'en',
        smart_format: true,
        filler_words: false,
        // interim_results must be true to use utterance_end_ms
        interim_results: true,
        utterance_end_ms: 1200,
        model: "enhanced",
        encoding: "linear16",
        sample_rate: 16000,
        channels: 1,
        diarize: true
    });
    ```

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jkroll-deepgram** [comment]
Hi @diegoasua, are you able to provide any request IDs from your testing? That will enable us to look deeper at the details of your requests to see if we can suggest any tweaks to improve your results.

That being said, it sounds like your testing is targeting proper names, which can be more challenging. For what it's worth, I just ran some quick tests using our [streaming test suite](https://github.com/deepgram/streaming-test-suite/tree/main) and my local mic. Deepgram was reliably able to transcribe `Craigslist` and `Amazon`, though it did struggle with James Corman as `James Cor`. Surnames in particular are difficult because they can be out-of-vocabulary for our default models, but when I added the parameter `keywords=corman`, then I was able to reliably transcribe `James Corman` too.

Glad to hear that you're enjoying other aspects of our developer experience, and based on the results you described, I do think we can find ways to yield better accuracy than you've initially seen. One other small point, I see you're testing our Enhanced model - please give [Nova-2](https://deepgram.com/learn/nova-2-speech-to-text-api) a try as well (`model: "nova-2"`), as it's our newest and best-performing model.

**Comment by diegoasua** [comment]
I did some digging and found

1. Transcription is much worse with AirPods on than using the Mac or iPhone microphone.
2. Clear voice and slow pace transcribes better than "natural" pace.
3. Slight minor accents drop performance a lot.
4. When speaking softly it performs worse. Louder seems to increase accuracy. 

Can you share data like [this](https://deepgram.com/learn/nova-2-speech-to-text-api) but for streaming? All the results shown there are for offline models. However it is not a fair comparison when performing real time inference since the model window has no access to the whole audio, so it does not have as much context from language and has to rely more on sounds, less on context. I would like to see error per rate of Google STT vs Deepgram streaming. Would be great if you can share some data on that.

Most of my tests were with the enhanced model but I have since switched to `nova-2` hoping it will ameliorate some of these issues.

Edit: After some tests I found `nova-2` streaming performs much better than the `enhanced` model, as long as I talk loud and clear.

**Reply by jkroll-deepgram** [reply]
Hi @diegoasua, thanks for sharing these insights, and I'm glad to hear that you've seen better results using a laptop or phone mic, and using the Nova-2 model.

If you look at our [Nova-2 announcement](https://deepgram.com/learn/nova-2-speech-to-text-api), we do a variety of competitor benchmarking. One graph does compare streaming word error rate (WER). See section ["The undisputed leader in real-time accuracy"](https://deepgram.com/learn/nova-2-speech-to-text-api#the-undisputed-leader-in-realtime-accuracy) (and copied graph below). We benchmark on a large, varied dataset across multiple domains. Note that several other vendors don't support streaming, so we are unable to include them in our streaming-specific analysis. We found that Nova-2 has a streaming WER of 10.7%, and Google V2 (Latest Long) has a streaming WER of 17.3%.

For even more robustness, you may also be interested in our `nova-2-phonecall` model for lower-quality audio, such as from phone calls with more background noise and lower bandwidth. In contrast, our `nova-2` model is trained on more general-purpose audio.

![image](https://github.com/deepgram/community/assets/135150417/64e53981-70a6-4939-bdd6-02a9221aaa58)

**Comment by diegoasua** [comment]
Thank you for the data Julia - did not see that graph until you pointed out. Exactly what I was looking for.

I gave `nova-2-phonecall` a try but had a similar or worse performance than `nova-2`. But with a few tweaks and speaking loudly it seems `nova-2` is performing much better now even with ambient noise.
